src/runtime/internal/string_storage.h

```diff
@@ -31,9 +31,26 @@ struct StringUtils {
         return count;
     }
 
-    static size_t count_length(const char *str) {
+    // retuns true if s1 contains s2 (within n characters)
+    static bool contains(const char *s1, const char *s2, size_t n) {
+        if (is_empty(s2)) { return true; }  // s2 is empty ... return true to match strstr
+        char starts_with = *s2;
+        for (size_t length = strlen(s2); length <= n; n--, s1++) {
+            if (*s1 == starts_with) {
+                for (size_t i = 1; i <= length; i++) {
+                    if (i == length)
```

Comment:
All if statements should be {braced} (I'm surprised that clang-format didn't complain, maybe we aren't running it on the runtime stuff? Perhaps please just manually format all these files)

Reply:
Oops ... yeah strange that clang-format didn't catch these ... it's certainly running on all the internal classes I added.  I'll go ahead and manually reformat this section!